java/java.nativeimage.debugger/src/org/netbeans/modules/java/nativeimage/debugger/api/ProcessInfo.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.java.nativeimage.debugger.api;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Information about a running process.
+ *
+ * @since 0.3
+ */
+public final class ProcessInfo {
```

Comment:
This might better belong into `extexecution` API. It is a generic functionality used to talk to executed processes.

Reply:
Well, thinking about this more, I'd remove it from the API and duplicate the classes instead. This will get obsolete when we abandon JDK 8. There is a `ProcessHandle` API since JDK 9.

Reply:
The class is duplicated instead of creating an API.
Duplication is documented in both classes and a note about future replacement with `ProcessHandle` is added.